src/ziggurat/middleware/stream_joins.clj

```diff
@@ -0,0 +1,21 @@
+(ns ziggurat.middleware.stream-joins
+  (:require [ziggurat.middleware.default :as dmw]))
+
+(defn- deserialize-stream-joins-message
+  "This function takes in the message(proto Byte Array) and the proto-class and deserializes the proto ByteArray into a
+  Clojure PersistentHashMap.
+  Temporary logic for migration of services to Ziggurat V3.0
+    If the message is of type map, the function just returns the map as it is. In older versions of Ziggurat (< 3.0) we stored
+    the messages in deserialized formats in RabbitMQ and those messages can be processed by this function. So we have this logic here."
+  [message proto-class topic-entity-name]
+  (reduce
```

Comment:
Hey, why do we do a reduce here? Won't the message have this structure:- 
`{:topic-key1 "some_byte_array" :topic-key2 "some_byte_array"}`

Reply:
Yes :) it will have that structure.  reduce is in case we have to do multiple joins (two or more), for now its only one...